keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -193,31 +218,65 @@ for new profile sources to be added in the future (e.g. Kubernetes predefined pr
+synchronised across the all nodes. 
+
+Users can build their own solutions to keep profiles in sync across nodes (e.g. 
+[using daemonsets](https://gardener.cloud/050-tutorials/content/howto/secure-seccomp/)). Or use 
+community driven projects (e.g. [seccomp config](https://github.com/UKHomeOffice/seccomp-config), 
+[openshift's machine config operator](https://github.com/openshift/machine-config-operator), 
+[seccomp operator](https://github.com/saschagrunert/seccomp-operator)) that focuses on solving similar 
+problems.
+
+Promoting LocalhostProfile to GA signals to the community that this is a feature they can use and rely on, 
+until a better option materializes. If and when that happens, a new KEP will be created accordingly.
+
 
-Drop the LocalhostProfile *string field. Keep the SeccompProfileLocalhost SeccompProfileType, but
-optionally change its value to LocalhostDeprecated.
+##### RuntimeProfile
 
-When creating a pod, the profile type can only be set to "Localhost" if the annotation is also set
-to localhost. When the kubelet goes to enforce the localhost profile, it fetches the path from the
-annotation.
+We propose maintaining the support to a single runtime profile, which will be defined by using the 
+SeccompProfileRuntime SeccompProfileType. The reasons being:
 
-The one gotcha is how to handle annotation update in this case. I'm tempted to say allow the update,
-and if the kubelet goes to enforce it and the annotation isn't set to localhost anymore, just treat
-it as an invalid localhost path (fail the pod).
-<<[/UNRESOLVED]>>
+- No changes to the current behavior. Users are currently not allowed to specify other runtime profiles.
+The existing API server rejects runtime profile names that are different than `runtime/default`.  
+The only exception being `docker/default` - for backwards compatibility. 
+- Most runtimes only support the default profile, although the CRI is flexible enough to allow the kubelet 
+to send other profile names. 
+- Dockershim does not currently provide a way to pass other runtime profile names.
+- Multiple runtime profiles has never been requested as a feature.
 
-<<[UNRESOLVED]>>
-What to do with RuntimeProfile field? There aren't any runtimes that support multiple built in
-profiles, and this feature has never been requested.
+If built-in support for multiple runtime profiles is needed in the future, a new KEP will be created to 
+cover its details. The implementation could be backwards compatible by creating a new `RuntimeProfile` field
+and setting its value to "default" whilst observing its immutability. 
 
-If we dropped this field for now (just assume it's runtime/default), how bad would it be to add it
-back at some point in the future if it was needed? As long as we defaulted the new field to
-"default" and it's immutable, it doesn't seem like it would be that problematic? Or am I forgetting
-something?
-<<[/UNRESOLVED]>>
 
 #### PodSecurityPolicy API
```

Comment:
@tallclair - I didn't expect PSP changes for this, given the trajectory of the PSP type.

Reply:
Since the PSP is not (yet) deprecated, I think it's important that it keeps up with the current API.

Reply:
would PSPs with existing seccomp defaultProfileName/allowedProfileNames annotations be effective at safeguarding the fields on pods (I would expect so)? if so, what is the benefit of adding complexity of parallel fields to PSPs if their current API expression covers the function we need?

Reply:
If we apply the PSP annotations to the new pod fields it should be just as effective.
But the lack of asymmetry may not give end users confidence on PSP/seccomp integration.
That said, I would probably suggest we update the PSP API until we find a substitute for it.